src/python/pants/engine/build_files.py

```diff
@@ -103,18 +90,19 @@ def resolve_unhydrated_struct(address_mapper, address):
   addresses = address_family.addressables
   if not struct or address not in addresses:
     _raise_did_you_mean(address_family, address.target_name)
+  address = next(build_address for build_address in addresses if build_address == address)
```

Comment:
This reads very weird. For normal definitions of `==`, this essentially reads as `address = address`. What's going on here? It's probably worth a comment.

Reply:
Agreed. This was preserved from before, but it is effectively: get the `BuildFileAddress` for this `Address`. Relates to #6657.